File: VB/XtraReport1.vb
Imports DevExpress.XtraReports.UI

Public Class XtraReport1

    ' Set the label's text and bookmark (this label is located on the PageHeader
    ' band, therefore it will show a bookmark for every report page).
    Private Sub xrLabel2_PrintOnPage(ByVal sender As System.Object, ByVal e As PrintOnPageEventArgs) _
    Handles xrLabel2.PrintOnPage
        Dim s As String = "Page " + (e.PageIndex + 1).ToString()
        CType(sender, XRLabel).Bookmark = s
        CType(sender, XRLabel).Text = s
    End Sub

End Class

File: CS/XtraReport1.cs
using System;
using DevExpress.XtraReports.UI;
// ...

namespace docBookmarks {
    public partial class XtraReport1 : XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        private void xrLabel2_PrintOnPage(object sender, PrintOnPageEventArgs e) {
            // Obtain the current page number.
            string s = (e.PageIndex + 1).ToString();

            // Set the label's text and bookmark (this label is located on the PageHeader
            // band, therefore it will show a bookmark for every report page).
            ((XRLabel)sender).Bookmark += s;
            ((XRLabel)sender).Text += s;
        }

    }
}
